toolchain/driver/driver.cpp

```diff
@@ -644,31 +646,36 @@ auto Driver::Compile(const CompileOptions& options) -> bool {
       }
     }
   });
+  RunResult result = {.success = true};
   for (const auto& input_file_name : options.input_file_names) {
     units.push_back(
         std::make_unique<CompilationUnit>(this, options, input_file_name));
+    result.per_file_success[input_file_name] = true;
```

Comment:
I wonder if it'd be better to represent this as an array of {filename, success} pairs instead of a string map -- a duplicate filename might be passed to the driver, so using filename as the primary key seems suspect. An array would also be lower overhead than a stringmap for non-test uses. (Using a string map for the file_test version still makes sense, though, I think.)

Reply:
Done.